src/main/java/de/tum/in/www1/artemis/config/migration/MigrationIntegrityException.java

```diff
@@ -0,0 +1,5 @@
+package de.tum.in.www1.artemis.config.migration;
+
+public class MigrationIntegrityException extends Exception {
```

Comment:
```suggestion
public class MigrationIntegrityException extends RuntimeException {
```
By using a runtime exception the log on startup directly shows the `MigrationIntegrityException` + stacktrace as cause for the startup failure. Otherwise the exception gets wrapped in another one which makes it less clear what happened.

The current output is similar to
```
java.lang.reflect.UndeclaredThrowableException: Failed to invoke event listener method
HandlerMethod details: 
Bean [MigrationRegistry]
Method [public void execute(...) throws MigrationIntegrityException]
Resolved arguments: 
[0] [type=org.springframework.boot.context.event.ApplicationReadyEvent] [value=org.springframework.boot.context.event.ApplicationReadyEvent[source=org.springframework.boot.SpringApplication@408b35bf]]

[... stack trace omitted ...]

Caused by: MigrationIntegrityException
[... another stack trace ...]
```

Reply:
Perfect, thanks.